Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -36,6 +36,8 @@ internal struct RawSyntaxData {
     var byteLength: Int
     /// Number of nodes in this subtree, excluding this node.
     var descendantCount: Int
+    /// Whether the tree contained by this layout has any missing or unexpected nodes.
+    var hasError: Bool
```

Comment:
Could we model this a `OptionSet` backed by a e.g.`UInt8`? I feel we will need other cached flags in the future.

Reply:
Good idea. I thought we could do this once we add more flags but it’s a good idea to do it now.